lib/collection/src/shards/shard_holder/mod.rs

```diff
@@ -208,6 +207,17 @@ impl ShardHolder {
                 .add(*shard_id);
         }
 
+        // Restore resharding hash ring if resharding is active and haven't reached
+        // `WriteHashRingCommitted` stage yet
+        if let Some(state) = self.resharding_state.read().deref() {
+            if state.stage < ReshardStage::WriteHashRingCommitted {
+                rings
+                    .get_mut(&state.shard_key)
+                    .expect("must have hash ring for current resharding shard key")
+                    .start_resharding(state.shard_id, state.direction);
+            }
+        }
```

Comment:
I don't think this is good enough in case of sharding down.

Imagine the following situation:
- sharding down
- just committed write hash ring
- shard still exists because we only drop it at the very end of resharding

If we now rebuild the hash ring it wrongfully includes the last shard we'll be dropping at the end of resharding.